### DISCUSSION TITLE
feedback(list-item): start slot doesn't have default padding.

### DISCUSSION THREAD

**Discussion by VandeurenGlenn** [material-components/material-web#4801] [Ideas]
I forgot to set data-variant on md-list-item so it renders start without padding, maybe better to add the data-variant="icon" behavior as the default ?
Spec says padding is always 16 except for video

Feels more naturally I guess when only needing to add data-variant on special use cases.

<img width="445" alt="image" src="https://github.com/material-components/material-web/assets/8685280/c2c47868-8925-4087-adff-5d9effd22012">

**Comment by e111077** [comment]
This was surfaced in the Lit Discord. Concerns here are for people slotting in other things like checkboxes or radios which are not icons. The pros for the current approach are that they fail fast and fail in a visually obvious manner of "this looks wrong", meaning that it indicates to the user that they should consider what the padding should be here since we do not have design tokens for them.

On the other hand, if this is indeed an overtly popular use case, there is merit to making it the default. The problem is that right now we don't have any data on what the most common use cases for list-item, menu-item, and select-option are.

At the peril of selection bias, please chime into this discussion with your thoughts, support, or opposition.

**Comment by VandeurenGlenn** [comment]
> Concerns here are for people slotting in other things like checkboxes or radios which are not icons.

@e111077 
But aren't those elements expected to have no padding or margin like an icon?
Unless you mean setting the data-variant=icon as default, but wouldn't do that, would just set the padding to standard 16 for slotted elements that don't have data-variant set

**Reply by e111077** [reply]
we would not want to do that because Design team has not finalized layout tokens and so far they have not tokenized padding for list item, and currently they have only tokenized icons videos avatars and images. And when they do, we have to support their new tokens and migrate users to the new padding tokens

What would you think of us adding `::slotted(md-icon,[data-variant=icon])` as a selector, so that if you're using our components, it should just work? this means if users use their own icons that are not `md-icon` they have to set `data-variant=icon`

**Reply by VandeurenGlenn** [reply]
> What would you think of us adding `::slotted(md-icon,[data-variant=icon])`

That would also be a nice thing to have for other elements like md-*insertvarianthere*-button.

**Reply by e111077** [reply]
chatting internally what do you think about

```html
<slot name="start">
  <slot name="start-icon"></slot>
  <slot name="start-image"></slot>
  <slot name="start-video"></slot>
  <slot name="start-avatar"></slot>
</slot>
```

Where default styling is applied to each specialized slot and nothing is added to "start"

**Reply by VandeurenGlenn** [reply]
That's perfect! Best of both worlds!

**Reply by e111077** [reply]
I'll work on this feature before 1.0 – it is going to require a gnarly migration of internal users

**Reply by e111077** [reply]
Follow #4925

Cleanup is here #4926 after we migrate our users
